### ISSUE TITLE
Is the TPE algo available in ML.NET?

### ISSUE THREAD

**Issue by leidadpig** [dotnet/machinelearning#4471]
Is the TPE algo available in ML.NET? Or is there any library available which implement TPE algo in C# to tune hyper params? Thanks!

**Comment by harishsk** [comment]
Sorry, there is no support for the TPE in ML.NET. As commented above, we would like to hear your use case though.

**Comment by leidadpig** [comment]
Thank both of you, I think AutoML.net is want I want.

**Comment by leidadpig** [comment]
I use TPE to tune hyper params a lot in Python, but due to current work which require C#  a lot, I have to train and use my model only in C#, it would be much appreciated if TPE is included in ML.NET.  Maybe I can train the model in Python and then use ONNX to convert it

**Comment by leidadpig** [comment]
I go through the AutoML's doc, but still confused about how it tune the hyper params, is grid search or other method used?

**Comment by leidadpig** [comment]
Or is it possible that use ML.Net to load a pretrained model trained by LightGBM python library so that no need to tune the hyper params in C#

**Comment by harishsk** [comment]
You can use Python to train your model, save it Onnx using Python and then use ML.NET to load and run inference on the model.

**Comment by leidadpig** [comment]
> You can use Python to train your model, convert it to Onnx using Python and then use ML.NET to load and run inferencing on the model.

Thanks

**Comment by justinormont** [comment]
> I go through the AutoML's doc, but still confused about how it tune the hyper params, is grid search or other method used?

We are using SMAC for hyperparameter optimization. We are optimizing three trainers in parallel, each has a warm-up of [20 random iterations](https://github.com/dotnet/machinelearning-automl/blob/40496fdf1ebdad9947e7db238868a42cdccb705b/src/Microsoft.ML.Auto/Sweepers/SmacSweeper.cs#L36-L37).

Basic sweep/search strategy:
* Test default hyperparameters on each trainer for the task (power of defaults) -- iteration 1 to ~10
  * Initial ordering is based on speed and general accuracy of the trainers, incase time expires before finishing all trainers; also allows ties to sway to the generally faster/better trainer
* _Cull to top three trainers_
* Hyperparameter optimization
  * Random sweeping as warm-up for SMAC -- iteration ~11 to ~70 (3 trainers * 20 iterations)
  * SMAC --  iteration 71+ (until time expires)

**Comment by leidadpig** [comment]
> > I go through the AutoML's doc, but still confused about how it tune the hyper params, is grid search or other method used?
> 
> We are using SMAC for hyperparameter optimization. We are optimizing three trainers in parallel, each has a warm-up of [20 random iterations](https://github.com/dotnet/machinelearning-automl/blob/40496fdf1ebdad9947e7db238868a42cdccb705b/src/Microsoft.ML.Auto/Sweepers/SmacSweeper.cs#L36-L37).
> 
> Basic sweep/search strategy:
> 
> * Test default hyperparameters on each trainer for the task (power of defaults) -- iteration 1 to ~10
>   
>   * Initial ordering is based on speed and general accuracy of the trainers, incase time expires before finishing all trainers; also allows ties to sway to the generally faster/better trainer
> * _Cull to top three trainers_
> * Hyperparameter optimization
>   
>   * Random sweeping as warm-up for SMAC -- iteration ~11 to ~70 (3 trainers * 20 iterations)
>   * SMAC --  iteration 71+ (until time expires)

Thank you very much

**Comment by harishsk** [comment]
@leidadpig I am assuming your questions have been answered and am closing the issue. Please feel free to reopen if necessary. Thanks.
